### PR TITLE
Feat(config) Modernize api config

### PR DESCRIPTION
**Pull request by Vince-janv**
## Description
The way the cg app is configured is very old fashioned. This causes instantiation of services used by the app to be very cumbersome (see `ext.py` and `app.py`).

This PR add pydantic-basesettings and utilises it to simplify the configuration. This in preparation for adding the freshdesk client.

### Added

- app_config.py containing the BaseSettings class AppConfig

### Changed

- keys in the app.config are now in snake_case to follow the normal use of BaseSettings 

### Fixed

- Unused config entries are removed

### CODE REVIEW

**Review by Vince-janv — COMMENTED** [review]
Removed some config settings that were not used

File: cg/server/app_config.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from pydantic_settings import BaseSettings
+
+
+class AppConfig(BaseSettings):
```
Comment (ChrOertlin) [inline cg/server/app_config.py:4]: I think you might need to specify the ENV file you get the values from, right? 

i.e. add a class Config in here that parses the env file:
```suggestion
class AppConfig(BaseSettings):
    ...
    ...
    class Config:
        env_file = str(ENV_FILE)

```

File: cg/apps/osticket.py [hunk 2]
```diff
@@ -26,10 +26,10 @@ def __init__(self):
     def init_app(self, app: Flask):
         """Initialize the API in Flask"""
         self.setup(
-            api_key=app.config["OSTICKET_API_KEY"],
-            domain=app.config["OSTICKET_DOMAIN"],
-            osticket_email=app.config["SUPPORT_SYSTEM_EMAIL"],
-            email_uri=app.config["EMAIL_URI"],
+            api_key=app.config["osticket_api_key"],
```
Comment (ChrOertlin) [inline cg/apps/osticket.py:29]:
```suggestion
            api_key=app.config.osticket_api_key,
```
Does this not work as well?

**Review by ChrOertlin — APPROVED** [review]
Nice modernisation!

File: cg/server/app_config.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    gunicorn_workers: int = 4
+    gunicorn_threads: int = 4
+    gunicorn_bind: str
+    gunicorn_timeout: int = 400
+    cg_sql_database_uri: str
+    cg_secret_key: str = "thisIsNotASafeKey"
+    invoice_max_price: int = 750000
```
Comment (henrikstranneheim) [inline cg/server/app_config.py:11]:
```suggestion
    invoice_max_price: int = 750_000
```

**Review by henrikstranneheim — APPROVED** [review]
Great 👍

File: cg/server/app_config.py (hunk 1 above)
Comment (Vince-janv) [reply cg/server/app_config.py:4]: BaseSettings takes the variables straight from the environment variables. But if you want to skip exporting them to the environment you can use the example you added to point to a file instead. 

Since the cg container has all variables exported to it I didn't add the Config class since it will not currently be used in production. I think it can easily be added if we decide to point to a file instead. 

Basically my reasoning is. Since we're not using it we can leave it out for now

File: cg/apps/osticket.py (hunk 2 above)
Comment (Vince-janv) [reply cg/apps/osticket.py:29]: Unfortunately `app.config` is just a dict that I update with the contents of `app_config`. Hence we can only access with values via square brackets

**Comment by Vince-janv** [comment]
StatusDB and order submission works in stage
